cloudformation/028-cfn-splash-page.yaml

```
@@ -0,0 +1,441 @@
+      Priority: 270
+
+  taskDef:
+    Type: AWS::ECS::TaskDefinition
+    Properties:
+      ContainerDefinitions:
+          - Name: "splash-page"
+          Essential: true
+          Image: !Ref ImageUri
+          LinuxParameters:
+            # https://docs.docker.com/engine/reference/run/#runtime-privilege-and-linux-capabilities
+            Capabilities:
+              ## Fargate does not allow adding Linux Capabilities. :-(
+              # Add:
+              #   - "CHOWN"
+              #   - "SETGID"
+              #   - "SETUID"
+              #   - "NET_BIND_SERVICE"
+              # Drop:
+              #   - "ALL"
+              Drop:
+                - SETPCAP
+                - MKNOD
+                - AUDIT_WRITE
+                - NET_RAW
+                - DAC_OVERRIDE
+                - FOWNER
+                - FSETID
+                - KILL
+                - SYS_CHROOT
+                - SETFCAP
+          LogConfiguration:
+            LogDriver: awslogs
+            Options:
+              awslogs-region: !Ref AWS::Region
+              awslogs-group: !Ref logGroup
+              awslogs-stream-prefix: openieui
+          MemoryReservation: 512
+          PortMappings:
+            - ContainerPort: 443
+              HostPort: 443
+              Protocol: tcp
+          Privileged: false
+          ReadonlyRootFilesystem: false
+      Cpu: 4096
+      ExecutionRoleArn: !GetAtt taskExecutionRole.Arn
+      Memory: 24576
+      NetworkMode: "awsvpc"
+      RequiresCompatibilities: 
+        - FARGATE
+      TaskRoleArn: !GetAtt taskRole.Arn
+      Tags:
+        - Key: ProjectName
+          Value: !Ref ProjectName
+
+  fargateService:
+    Type: AWS::ECS::Service
+    Properties:
+      LaunchType: FARGATE
+      ServiceName: !Sub "${ServiceName}-1"
+      Cluster: !ImportValue prod-general-infra-EcsCluster
+      SchedulingStrategy: REPLICA
+      DesiredCount: !Ref DesiredServices
+      TaskDefinition: !Ref taskDef
+      HealthCheckGracePeriodSeconds: 90
+      NetworkConfiguration:
+        AwsvpcConfiguration:
+          AssignPublicIp: DISABLED
+          Subnets: 
+            - Fn::Select:
+              - 0
+              - Fn::Split:
+                - ','
+                - Fn::ImportValue: prod-vpc-PrivateSubnetIds
+            - Fn::Select:
+              - 5
+              - Fn::Split:
+                - ','
+                - Fn::ImportValue: prod-vpc-PrivateSubnetIds
+          SecurityGroups:
+            - !Ref fargateSecurityGroup
+            - !ImportValue "prod-vpc-Ec2EcsHostDefaultSecurityGroup"
+      LoadBalancers:
+        - TargetGroupArn: !Ref targetGroupExternalApi
+          ContainerPort: 80
+          ContainerName: auth-sidecar
+        - TargetGroupArn: !Ref targetGroupExternalBrowser
+          ContainerPort: 8182
+          ContainerName: !Ref ServiceName
+        - TargetGroupArn: !Ref internalTargetGroup
+          ContainerPort: 8182
+          ContainerName: !Ref ServiceName
```

```suggestion
        - TargetGroupArn: !Ref targetGroupExternalApi
          ContainerPort: 80
          ContainerName: !Ref ServiceName
```
We're only running the one service here